### DISCUSSION TITLE
increase number of sources?

### DISCUSSION THREAD

**Discussion by ram-sh** [PromtEngineer/localGPT#77] [General]
Hello, 

Awesome project! Thanks for sharing!

I had 20,000 text files embedded in my chroma db, and each file is a short story, i currently use `Wizard-Vicuna-13B-Uncensored-HF` as my model, and I wanted to know how to change the model to use more than just 4 sources in the creation of a new story?  or perhaps I approach this the wrong way? 

thanks again for your work!!
Best,
RS

**Comment by PromtEngineer** [comment]
In the retreiver object, you can set the number of documents you want to use as top_k, here is the link: https://python.langchain.com/en/latest/reference/modules/retrievers.html#langchain.retrievers.ChatGPTPluginRetriever.top_k

**Reply by LazyCat420** [reply]
Was wondering where is the retriever object located or how can you identify it? found the retriever function in run_localGPT_API.py but couldn't figure out how to get it to work.

**Comment by gulzatmamytbek1** [comment]
I would like to know this too. Good question

**Comment by PromtEngineer** [comment]
You can change the number of sources here: https://github.com/PromtEngineer/localGPT/blob/6f894706d9c371115d691f0d9e3b117fb42ba216/run_localGPT.py#L190

By replacing it with:  `retriever = db.as_retriever(search_kwargs={"k": 5})`
